packages/api/src/providers/provider.ts

```diff
@@ -17,6 +17,7 @@ export enum ConsumerHealthDataType {
 export type DAPIParams = {
   timezoneId?: string;
   accessToken?: string;
+  patientId?: string;
```

Comment:
On DAPI, we call them `user`, leaving `patient` to MAPI. Can we change that here and everywhere this PR refers to patient?

Reply:
Taken to slack considering the nature of the discussion: https://metriport.slack.com/archives/C04DMKE9DME/p1692829416471799

Reply:
Solution: remove `patientId` and use `deviceUserId` instead, and not as an extraParam, but as an attribute in `ProviderMapItem`